src/ReportManager/EFBForm/FormSchemaContext/index.js

```diff
@@ -0,0 +1,29 @@
+import React, { createContext } from 'react';
+
+export const FormSchemaContext = createContext(null);
+
+const FormSchemaContextProvider = ({ schema, onFieldChange, formData, children }) => {
```

Comment:
I like this idea! A way to provide friendly methods and hide the internals of the schema.

Just one thing, you are forgetting to provide the `order` UI schema property that tells the ordering of the sections: https://allenai.atlassian.net/wiki/spaces/ER/pages/30116741125/Form+Builder+Generated+JSON+Schemas#Order

Also, we don't need to keep the `{ json: {...}, ui: {...} }` format for the fields anymore. Since the responsibility here is just to render the form, we can merge them into a single object that contains all the information needed to render the component. The tricky part is that every field will have different properties, so we will need to clean one by one. But the good thing is that we already have this logic covered in the EFB: https://github.com/PADAS/report-form-builder/blob/master/app/components/EventTypeForm/utils/makeFieldsFromSchema/index.js#L40

An example of how that would look for a text field is:
```
// Outside of the component
const buildTextFieldObject = (fieldName, jsonSchema, uiSchema, isRequired) => ({
  defaultInput: jsonSchema.default,
  description: jsonSchema.description,
  inputType: uiSchema.inputType,
  isActive: !jsonSchema.deprecated,
  isRequired,
  label: jsonSchema.title,
  placeholder: uiSchema.placeholder,
  value: fieldName,
});

...

  const getFieldProps = (fieldName) => {
    const jsonProps = schema.json.properties[fieldName];
    const isRequired = schema.json.required.includes(fieldName);
    const uiProps = schema.ui.fields[fieldName];

    switch (uiProps.type) {
      case 'TEXT':
        return buildTextFieldObject(fieldName, jsonProps, uiProps, isRequired);
      ...
    }
  };
```

This way we match the field objects just like in the EFB reducing the complexity between the two repos! You can see the full lists of props that each field needs to have here: https://github.com/PADAS/report-form-builder/blob/master/app/components/EventTypeForm/FormBuilder/canvasFieldDetailsFactory/index.js

Reply:
Let me know what you think. If you like it, but you think it will take you long, we don't need to do it now, we can add each field type function as we work on that field type. But you can at least leave that switch with its // TODO